cpp/src/copying/contiguous_split.cu

```diff
@@ -0,0 +1,219 @@
+   {      
+      // there's some unnecessary recomputation of sizes happening here, but it really shouldn't affect much.
+      size_t data_size = 0;
+      size_t validity_size = 0;      
+      column_buf_size_functor_impl<T>{}(in, data_size, validity_size);
+
+      // outgoing pointers
+      char* data = dst;
+      bitmask_type* validity = validity_size == 0 ? nullptr : reinterpret_cast<bitmask_type*>(dst + data_size);
+
+      // increment working buffer
+      dst += (data_size + validity_size);      
+
+      // custom copy kernel (which should probably just be an in-place copy() function in cudf.
+      cudf::size_type num_els = cudf::util::round_up_safe(in.size(), cudf::experimental::detail::warp_size);
+      constexpr int block_size = 256;
+      cudf::experimental::detail::grid_1d grid{num_els, block_size, 1};      
+      
+      // so there's a significant performance issue that comes up. our incoming column_view objects
+      // are the result of a slice.  because of this, they have an UNKNOWN_NULL_COUNT.  because of that,
+      // calling column_device_view::create() will cause a recompute of the count, which ends up being
+      // extremely slow because a.) the typical use case here will involve huge numbers of calls and
+      // b.) the count recompute involves tons of device allocs and memcopies.
+      //
+      // so to get around this, I am manually constructing a fake-ish view here where the null
+      // count is arbitrarily bashed to 0.            
+      //
+      column_view   in_wrapped{in.type(), in.size(), in.head<T>(), 
+                               in.null_mask(), in.null_mask() == nullptr ? UNKNOWN_NULL_COUNT : 0,
+                               in.offset() };
+      mutable_column_view  mcv{in.type(), in.size(), data, 
+                               validity, validity == nullptr ? UNKNOWN_NULL_COUNT : 0 };      
+      if(in.nullable()){               
+         copy_in_place_kernel<block_size, T, true><<<grid.num_blocks, block_size, 0, 0>>>(
+                           *column_device_view::create(in_wrapped), 
+                           *mutable_column_device_view::create(mcv));         
+      } else {
+         copy_in_place_kernel<block_size, T, false><<<grid.num_blocks, block_size, 0, 0>>>(
+                           *column_device_view::create(in_wrapped), 
+                           *mutable_column_device_view::create(mcv));
+      }
+      mcv.set_null_count(cudf::UNKNOWN_NULL_COUNT);
+
+      out_cols.push_back(mcv);
+   }
+};
+
+template<>
+struct column_copy_functor_impl<string_view> {
+   void operator()(column_view const& in, char*& dst, std::vector<column_view>& out_cols) 
+   {
+      CUDF_FAIL("contiguous_split for strings not implemented yet");
+   };
+};
+
+struct column_copy_functor {
+   template<typename T>   
+   void operator()(column_view const& in, char*& dst, std::vector<column_view>& out_cols)
+   {
+      column_copy_functor_impl<T> fn{};      
```

Comment:
What's the purpose of `column_copy_functor_impl`? Seems like an unnecessary level of indirection. Instead, the body of `column_copy_functor_impl` could be moved here and the `column_copy_functor::operator()` could be specialized accordingly.

Reply:
My thinking here is that the implementation for string_view is going to be different.  It's going to have to allocate multiple buffers and distribute them to child columns_views.  This is the pattern I've seen when dealing with a situation like this (eg, Dave Wendt's string_view addition to copy_if_else).